src/pages/experiments/[experimentId]/plots-and-tables/volcano/index.jsx

```diff
@@ -124,25 +130,7 @@ const VolcanoPlot = ({ experimentId }) => {
     },
   ];
 
-  useEffect(() => {
-    if (!config) return;
-    setDataPointStatus();
-    const generatedSpec = generateSpec(config, plotData);
-    setSpec(generatedSpec);
-  }, [config]);
-
-  useEffect(() => {
-    if (data.length === 0) return;
-    setDataPointStatus();
-  }, [data]);
-
-  useEffect(() => {
-    if (plotData.length === 0) return;
-    const generatedSpec = generateSpec(config, plotData);
-    setSpec(generatedSpec);
-  }, [plotData]);
-
-  const setDataPointStatus = () => {
+  const calculateDataPointStatus = () => {
     const dataPoints = _.cloneDeep(data);
```

Comment:
Make `data` and `config` into a parameters for this function. That should turn this into a pure function. Then you should be able to move this function out of the React function body and save some CPU cycles.